test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+
+      speech.createRecognizeJob = function(config, callback) {
+        callback(error, null, RES);
+      };
+
+      speech.createDetectJob(FILE, REQ, function(err, response, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(response, null);
+        assert.strictEqual(apiResponse, RES);
+        done();
+      });
+    });
+
+    it('should return the resp from createRecognizeJob()', function(done) {
+      var RES1 = speech.operation({
+        name: '1234'
+      });
+      var RES2 = {
+        name: '1234'
+      };
+
+      speech.createRecognizeJob = function(config, callback) {
+        callback(null, RES1, RES2);
+      };
+
+      speech.createDetectJob(FILE, REQ, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert.equal(response.name, '1234');
+        assert.strictEqual(apiResponse, RES2);
+
+        done();
+      });
+    });
+  });
+
+  describe('createRecognizeStream', function() {
+    var REQ = {};
+
+    it('should make the correct API request', function(done) {
+      var request = REQ;
+      var mockStream = {
+        write: function(data) {
+          assert(data);
+          assert.strictEqual(data, REQ);
+          setImmediate(function() {
+            done();
+          });
+        }
+      };
+
+      speech.requestWritableStream = function(protoOpts) {
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'streamingRecognize');
+
+        return mockStream;
+      };
+
+      assert.strictEqual(speech.createRecognizeStream(request), mockStream);
+    });
+  });
+
+  describe('createDetectStream', function() {
+    var REQ = {};
+
+    it('should make the correct API request', function(done) {
+      var expected = 'how old is the Brooklyn Bridge';
+      var filePath = '../../system-test/data/bridge.raw';
+      var audioFile = fs.readFileSync(path.join(__dirname, filePath));
+      var request = REQ;
+      var upstreamWriteCount = 0;
+      var mockStream = {
+        write: function(data) {
+          upstreamWriteCount++;
+          assert(data);
+          if (upstreamWriteCount === 1) {
+            assert.deepEqual(data, {
+              streamingConfig: REQ
+            });
+            assert.strictEqual(data.streamingConfig, REQ);
+          } else if (upstreamWriteCount === 2) {
+            assert(data.audioContent);
+          }
+        },
+        on: function(event, handler) {
+          if (event === 'response') {
+            this.responseHandler = handler;
+          }
+        },
+        emit: function(event, data) {
+          if (event === 'data') {
+            this.pipeTo.write(data);
+          } else if (event === 'response') {
+            this.responseHandler(data);
+          }
+        },
+        once: function() {},
+        pipe: function(stream) {
+          this.pipeTo = stream;
+        }
+      };
+
+      speech.requestWritableStream = function(protoOpts) {
```

Comment:
I don't think we need to stub this, but rather we just need to stub `createRecognizeStream` and make sure the reqOpts are what we expect.